presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -247,9 +258,14 @@ private synchronized void refreshNodesInternal()
             }
         }
 
-        allNodes = new AllNodes(activeNodesBuilder.build(), inactiveNodesBuilder.build(), shuttingDownNodesBuilder.build());
+        AllNodes allNodes = new AllNodes(activeNodesBuilder.build(), inactiveNodesBuilder.build(), shuttingDownNodesBuilder.build(), coordinatorsBuilder.build());
+        this.allNodes = allNodes;
         activeNodesByConnectorId = byConnectorIdBuilder.build();
         coordinators = coordinatorsBuilder.build();
+
+        // notify listeners of the new node classification
```

Comment:
`new node classification`? Isn't this just notifying people of *any* change to AllNodes?